### PR TITLE
Feature/add search

### PR DESCRIPTION
**Pull request by duytang**
Feature Search Video

### CODE REVIEW

File: YouTube/Classes/Base/BaseNavigationController.swift [hunk 1]
```diff
@@ -10,13 +10,31 @@ import UIKit
 
 class BaseNavigationController: UINavigationController {
 
+    override init(rootViewController: UIViewController) {
+        super.init(rootViewController: rootViewController)
+        self.view.backgroundColor = UIColor.clearColor()
```
Comment (phuongvnc) [inline YouTube/Classes/Base/BaseNavigationController.swift]: remove `self`

File: YouTube/Classes/CustomViews/ResultSearchCell/ResultSearchCell.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+//
+//  Created by Duy Tang on 8/24/16.
+//  Copyright © 2016 Duy Tang. All rights reserved.
+//
+
+import UIKit
+
+class ResultSearchCell: BaseTableViewCell {
+
+    @IBOutlet weak private var nameVideoLabel: UILabel!
```
Comment (at-phuongvnc) [inline YouTube/Classes/CustomViews/ResultSearchCell/ResultSearchCell.swift]: missing `private`

File: YouTube/Classes/ViewControllers/SearchViewController/SearchViewController.swift [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+            return
+        }
+        isLoading = true
+        showLoading()
+        var parameters = [String: AnyObject]()
+        parameters["part"] = "snippet"
+        parameters["maxResults"] = Options.MaxResult
+        parameters["pageToken"] = nextPage
+        parameters["q"] = key
+        MyVideo.searchVideoForKey(parameters) { (success, response, nextPageToken, error) in
```
Comment (phuongvnc) [inline YouTube/Classes/ViewControllers/SearchViewController/SearchViewController.swift]: change name of class

File: YouTube/Classes/ViewControllers/SearchViewController/SearchViewController.swift [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+// MARK:- Extension UITableViewDataSource SearchViewController
+extension SearchViewController: UITableViewDataSource {
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if tableView.tag == 0 {
```
Comment (phuongvnc) [inline YouTube/Classes/ViewControllers/SearchViewController/SearchViewController.swift:174]: return listKey.count ?? 0

File: YouTube/Classes/Video/MyVideo.swift [hunk 5]
```diff
@@ -62,5 +62,44 @@ class MyVideo {
...
+        APIRequest.GET(api, parameter: parameters, success: { (response) in
+            if let data = response as? [String: AnyObject], items = data["items"] as? NSArray,
+                item = items[0] as? [String: AnyObject],
+                snippet = item["snippet"] as? [String: AnyObject],
+                thumbnail = snippet["thumbnails"] as? [String: AnyObject],
+                imageChannel = thumbnail["high"] as? [String: AnyObject] {
+                    if let url = imageChannel["url"] as? String {
+                        completion(response: url)
+                    }
+            }
```
Comment (phuongvnc) [inline YouTube/Classes/Video/MyVideo.swift:103]: miss `else` of if

File: YouTube/Classes/ViewControllers/DetailFavoriteViewController/DetailFavoriteViewController.swift [hunk 6]
```diff
@@ -15,70 +15,220 @@ class DetailFavoriteViewController: BaseViewController {
     @IBOutlet weak private var nameListFavoriteLabel: UILabel!
     @IBOutlet weak private var listVideoFavoriteTableView: UITableView!
     var favorite = Favorite()
-    var videoFavorites: Results<VideoFavorite>?
     private struct Options {
-        static let HeightOfRow: CGFloat = 230
+        static let HeightOfRow: CGFloat = 205
+        static var cellSnapshot: UIView? = nil
```
Comment (phuongvnc) [inline YouTube/Classes/ViewControllers/DetailFavoriteViewController/DetailFavoriteViewController.swift]: don't need define it in Options

File: YouTube/Classes/ViewControllers/SearchViewController/SearchViewController.swift [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+        searchResultTableView.hidden = true
+    }
+    // MARK:- Set Up
+    override func setUp() {
+        modalPresentationStyle = .OverCurrentContext
+    }
+    // MARK:- Set Up UI
+    override func setUpUI() {
+        keySearchBar.backgroundImage = UIImage()
+        config()
```
Comment (phuongvnc) [inline YouTube/Classes/ViewControllers/SearchViewController/SearchViewController.swift]: what is config ?

**Comment by phuongvnc** [comment]
LGTM

**Pull request merged**